### PR TITLE
Fix the build failing on node v12

### PR DESCRIPTION
**Pull request by Page-**
Change-type: patch
Signed-off-by: Thodoris Greasidis <thodoris@balena.io>

Resolves: # 
HQ: <url> 
See: <url> 
Depends-on: <url> 
Change-type: major|minor|patch 

---
##### Contributor checklist

- [ ] Includes tests
- [ ] Includes typings
- [ ] Includes updated documentation
- [ ] Includes updated build output

### CODE REVIEW

File: gulpfile.coffee [hunk 1]
```diff
@@ -80,5 +76,8 @@ gulp.task 'pack-browser', ->
 		.pipe(minify())
 		.pipe(gulp.dest(OPTIONS.directories.build))
 
+gulp.task 'build',
+	gulp.series(['build-node', 'pack-browser'])
+
 gulp.task 'build-browser', (callback) ->
```
Comment (Page-) [inline gulpfile.coffee]:
```suggestion
gulp.task 'build-browser', gulp.series(['build-node', 'pack-browser'])
```

Comment (thgreasi) [reply gulpfile.coffee]: I just changed the other above before pushing. And as far as I can see this isn't used at all.

File: package.json [hunk 2]
```diff
@@ -62,13 +62,13 @@
...
     "dotenv": "^4.0.0",
     "dtslint": "^1.0.3",
-    "gulp": "^3.9.1",
-    "gulp-coffee": "^2.3.5",
+    "gulp": "^4.0.2",
+    "gulp-coffee": "^3.0.3",
     "gulp-coffeelint": "^0.6.0",
     "gulp-insert": "^0.5.0",
-    "gulp-mocha": "^4.3.1",
+    "gulp-mocha": "^7.0.2",
```
Comment (Page-) [inline package.json]: Needs to be 6.x as 7.x drops node 6 support (which is what the appveyor tests run with)

File: gulpfile.coffee (hunk 1 above)
Comment (Page-) [reply gulpfile.coffee]: I'm also fine with removing it completely :P

**Comment by thgreasi** [comment]
And now I'm a reviewer on my own commits...

File: tests/integration/models/application.spec.coffee [hunk 3]
```diff
@@ -76,6 +76,7 @@ describe 'Application Model', ->
 					balena.pine.get
 						resource: 'application'
 						options:
+							$filter: is_public: false
```
Comment (Page-) [inline tests/integration/models/application.spec.coffee]: Is this meant to be part of the PR @thgreasi ?

Comment (thgreasi) [reply tests/integration/models/application.spec.coffee]: Nope, let me split it, good catch. I just had to added for the tests to pass on staging and forgot to reset before pushing.

Comment (thgreasi) [reply tests/integration/models/application.spec.coffee]: Done

**Comment by thgreasi** [comment]
Indeed @sradevski , I would like to go for rollup as well.

**Pull request merged**